### DISCUSSION TITLE
Realtime with Multiplayer - post announcement discussion

### DISCUSSION THREAD

**Discussion by johnoscott** [supabase/supabase#6344] [Questions] [answered]
Excited for Realtime with Multiplayer. I am already following https://github.com/supabase/realtime 

I have some questions :
1. I signed up for early access ; what is the timeframe that you (hope) access will start ?
2. What is the CRDT implementation ? Did you use an existing OSS solution like Yjs ?

**Comment by w3b6x9** [comment, marked as answer]
@johnoscott CRDT implementation is called Observe-Remove-Set-Without-Tombstones (ORSWOT). Here's the [paper](https://gsd.di.uminho.pt/members/cbm/ps/delta-crdt-draft16may2014.pdf) if you wanted to dig through it. It's an integration with [Phoenix Presence](https://hexdocs.pm/phoenix/Phoenix.Presence.html) which is open-source.

We're currently investigating using the new Realtime broadcast functionality as the transport layer for Yjs.

**Reply by johnoscott** [reply]
@w3b6x9 so does that mean you are building your own CRDT implementation based on that paper or using Yjs ? 

Also, can you provide a (rough is fine) timeframe as to when the features will be ready ? I am trying to understand if Supabase Realtime with Multiplayer will fit the timeline of a project I am working on.

**Reply by bhvngt** [reply]
@w3b6x9. Would it be possible to share the rough timeline for the multiplayer features (Broadcast and presence). Like @johnoscott, I too have a particular use case where the multiplayer feature would help me a lot, so just checking if I can align my timeline.

**Reply by w3b6x9** [reply]
> @w3b6x9 so does that mean you are building your own CRDT implementation based on that paper or using Yjs ?

@johnoscott we integrated Phoenix Presence which is based on that paper I shared with you.

> Would it be possible to share the rough timeline for the multiplayer features (Broadcast and presence).

@johnoscott @bhvngt we were opening it up slowly to developers but have paused in order to prioritize adding multi-tenancy to Realtime. I don't have a clear timeline for you at this time. We hope to continue opening up Broadcast + Presence in the next few weeks.

**Reply by johnoscott** [reply]
Ok thanks. Looking forward to hearing more.

**Reply by bhvngt** [reply]
Thanks @w3b6x9. Looking forward to this feature set

**Reply by RickyvdBerg** [reply]
Just chiming in to say I am also looking forward to using this feature!

**Reply by jonas-kgomo** [reply]
Looking forward to this feature. Indeed

**Reply by RickyvdBerg** [reply]
Will we have control over some of the server code with the new realtime API? This would be essential for multiplayer games where you don't trust the client? Also what would the pricing/scaling of this look like. Since websockets don't really like to scale horizontally, I'm curious as to what kind of concurrent users we can expect to support with the different pricing plans. @w3b6x9
